src/main/java/io/spokestack/spokestack/tts/SpokestackTTSClient.java

```diff
@@ -0,0 +1,155 @@
+package io.spokestack.spokestack.tts;
+
+import com.google.gson.Gson;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+
+import java.util.HashMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Spokestack TTS client.
+ *
+ * <p>
+ * The TTS client communicates with the Spokestack TTS service using simple REST
+ * requests, receiving audio URLs that can be played with the Android media
+ * player.
+ * </p>
+ */
+public final class SpokestackTTSClient {
+    private String ttsUrl = "https://core.pylon.com/speech/v1/tts/synthesize";
+    private static final MediaType APPLICATION_JSON =
+          MediaType.parse("application/json; charset=utf-8");
+    private final OkHttpClient httpClient;
+    private final TTSCallback ttsCallback;
+    private final Gson gson;
+    private String ttsApiKey;
+
+    /**
+     * Create a new Spokestack TTS client. This client should be created once
+     * and reused, as it contains an HTTP client with internal connection and
+     * thread pools.
+     *
+     * @param callback The callback object used to deliver an audio URL when it
+     *                 becomes available.
+     */
+    public SpokestackTTSClient(TTSCallback callback) {
+        this(callback,
+              new OkHttpClient.Builder()
+                    .connectTimeout(5, TimeUnit.SECONDS)
+                    .readTimeout(5, TimeUnit.SECONDS)
+                    .build()
+        );
+    }
+
+    /**
+     * Create a new Spokestack TTS client with an API key and a provided HTTP
+     * client. Used for testing.
+     *
+     * @param callback The callback object used to deliver an audio URL when it
+     *                 becomes available.
+     * @param client   The HTTP client to use for requests.
+     */
+    public SpokestackTTSClient(TTSCallback callback,
+                               OkHttpClient client) {
+        this.gson = new Gson();
+        this.ttsCallback = callback;
+        this.httpClient = client;
+    }
+
+    /**
+     * Set the URL used to synthesize text.
+     * @param url The URL to which synthesis requests should be sent.
+     */
+    public void setTtsUrl(String url) {
```

Comment:
There's no need to expose this implementation detail (yet) IMO.

Reply:
The reason I did this is because we're releasing this while core is still active, but I don't think that will be the case forever. If/when core goes away, any projects dependent on the old version will be broken forever. Requiring someone to bump their dependency (which may or may not include other changes they don't want to deal with) because a URL has changed feels...unneighborly.